CMaxTable.lua

```diff
@@ -19,8 +19,7 @@ end
 
 function CMaxTable:updateGradInput(input, gradOutput)
    for i=1,#input do
-      self.gradInput[i] = torch.Tensor()
-      self.gradInput[i]:resizeAs(input[i]):fill(0.0)
+      self.gradInput[i] = input[i]:clone():fill(0.0)
```

Comment:
you can add something like
```lua
self.gradInput[i] = self.gradInput[i] or input[i].new()
self.gradInput[i]:resizeAs(input[i]):zero()
```
This way, you can change the size of the input during forward calls, which is not possible with current implementation